units/include/scipp/units/unit.tcc

```diff
@@ -193,6 +193,16 @@ Derived inverse_trigonometric(const Unit_impl<Derived> &a) {
       a.name() + ".");
 }
 
+template <class Derived>
+Derived inverse_trigonometric_binary(const Unit_impl<Derived> &a, const Unit_impl<Derived> &b) {
+  if ( a == b )
```

Comment:
Is this condition special to `atan2`? Are there other "inverse trigonometric binary" operations? If not, I suggest to rename this function (or just getting rid of it, since it would not be reused) and using `boost::units::atan2` instead of `asin`.